presentation/src/main/java/org/cryptomator/presentation/util/BiometricAuthenticationMigration.kt

```diff
@@ -0,0 +1,142 @@
+package org.cryptomator.presentation.util
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.security.keystore.KeyPermanentlyInvalidatedException
+import androidx.biometric.BiometricFragment
+import androidx.biometric.BiometricPrompt
+import androidx.biometric.FingerprintDialogFragment
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import org.cryptomator.domain.Vault
+import org.cryptomator.presentation.R
+import org.cryptomator.presentation.model.VaultModel
+import org.cryptomator.util.crypto.BiometricAuthCryptor
+import org.cryptomator.util.crypto.CryptoMode
+import org.cryptomator.util.crypto.UnrecoverableStorageKeyException
+import javax.crypto.BadPaddingException
+import timber.log.Timber
+
+class BiometricAuthenticationMigration(
+	private val callback: Callback, private val context: Context, private val useConfirmationInFaceUnlockAuth: Boolean
+) {
+
+	interface Callback {
+
+		fun onBiometricAuthenticationMigrationFinished(vaults: List<VaultModel>)
+		fun onBiometricAuthenticationFailed(vaults: List<VaultModel>)
+		fun onBiometricKeyInvalidated(vaults: List<VaultModel>)
+	}
+
+	private lateinit var promptInfo: BiometricPrompt.PromptInfo
```

Comment:
I'd suggest either converting all state objects to properties (i.e. `decryptedVaults`, `reEncryptedVaults`, etc.) or changing this to a parameter as well.

Reply:
`promptInfo` does not change over time whereas the other variables are used during recursion so I think it makes sense to keep it that way but I moved the initiation of the variable out of the method in https://github.com/cryptomator/android/commit/3f77326d72c5e78f45930aa83f46a11ced2e9123